Automattic/portkey-android
# Color picker v2: 1 - Add bottom sheet

First (big) step of #465. Replaces the color picker RecyclerView with an (empty) bottom sheet. I'll add the actual color picker UI in a subsequent PR, but the bottom sheet behavior alone ended up being a lot of (tricky) code.

![stories-bottom-sheet](https://user-images.githubusercontent.com/9613966/93965973-72266380-fd9e-11ea-8183-b3a762db35cb.gif)

I had to do some extra work to get the bottom sheet to play nice with the keyboard. Not only is there no easy way to hide the keyboard when the bottom sheet appears, but the layout recalculation when the keyboard disappears doesn't account for the bottom sheet correctly, so the text ends up being behind the bottom sheet sometimes.

The approach I went for is to manually trigger keyboard showing and hiding events when the bottom sheet is called/dismissed. Also, the view is adjusted to the bottom sheet's new height, and the height of the bottom sheet matches the keyboard's height to keep jank to a minimum.

Interested in any better ways there might be to approach all this, and more importantly if there are any breaking issues with the current behavior (at this point it's working pretty well for me).

### To test:
1. Open up the text editor, tap the color icon on the right
2. Observe that keyboard goes down, bottom sheet comes up
3. Try dismissing the bottom sheet in each of three ways:
  i. Press the color icon again
  ii. Swipe the bottom sheet down
  iii. Tap on the EditText
4. Observe that the bottom sheet goes away, and the keyboard comes back up
